packages/account/src/wallet/base-wallet-unlocked.ts

```diff
@@ -123,9 +126,14 @@ export class BaseWalletUnlocked extends Account {
    * @param transactionRequestLike - The transaction request to simulate.
    * @returns A promise that resolves to the CallResult object.
    */
-  async simulateTransaction(transactionRequestLike: TransactionRequestLike): Promise<CallResult> {
+  async simulateTransaction(
+    transactionRequestLike: TransactionRequestLike,
+    options?: EstimateTransactionParams
+  ): Promise<CallResult> {
     const transactionRequest = transactionRequestify(transactionRequestLike);
-    await this.provider.estimateTxDependencies(transactionRequest);
+    if (options?.estimateTxDependencies === undefined || options.estimateTxDependencies === true) {
+      await this.provider.estimateTxDependencies(transactionRequest);
+    }
```

Comment:
I believe we can change this to:

```ts
 async simulateTransaction (
    transactionRequestLike: TransactionRequestLike,
    options: EstimateTransactionParams = { estimateTxDependencies: true }
  ): Promise<CallResult> {
    const transactionRequest = transactionRequestify(transactionRequestLike);
    if (options.estimateTxDependencies) {
      await this.provider.estimateTxDependencies(transactionRequest);
    }
```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1767/commits/0e8501eddeec712c6889afabeeec618cd7a7d44a